src/api/worker/facades/UserManagementFacade.ts

```diff
@@ -394,20 +397,37 @@ export class UserManagementFacade {
 		const key = await this.groupManagement.getGroupKeyViaUser(mailGroupId, viaUser)
 		const mailboxGroupRoot = await this.nonCachingEntityClient.load(MailboxGroupRootTypeRef, mailGroupId)
 		if (mailboxGroupRoot.mailboxProperties == null) {
-			return this.createMailboxProperties(mailboxGroupRoot, key)
+			mailboxGroupRoot.mailboxProperties = await this.createMailboxProperties(mailboxGroupRoot, key)
 		}
-		return await this.nonCachingEntityClient.load(MailboxPropertiesTypeRef, mailboxGroupRoot.mailboxProperties, undefined, undefined, key)
+		const properties = await this.nonCachingEntityClient.load(MailboxPropertiesTypeRef, mailboxGroupRoot.mailboxProperties, undefined, undefined, key)
+		if (properties.mailAddressProperties.length === 0) {
+			return this.migrateFromOldSenderName(properties, viaUser)
+		}
+		return properties
+	}
+
+	private async migrateFromOldSenderName(mailboxProperties: MailboxProperties, viaUser: Id) {
```

Comment:
is this duplicated

Reply:
They are slightly different in a way where I would keep it this way